engine/swarm/secrets.md

```diff
@@ -0,0 +1,280 @@
+                            wordpress:latest
+    ```
+
+6.  Access `http://localhost:30000/` from any Swarm node and begin using
+    WordPress.
+
+## Example: Rotate a secret
+
+This example builds upon the previous one. In this scenario, you create a new
+secret with the new MySQL password, update the `mysql` and `wordpress` services
+to use it, then remove the old secret.
+
+1.  Create the new secret, and name it `mysql_password_2016`.
+
+    ```bash
+    $ openssl rand -base64 20 |docker secret create mysql_password_2016
+      lwxgks2rojdbvaui5kay1suvr
+    ```
+
+2.  Update the `mysql` service to use the new password, keeping the target path
+    at `/run/secrets/mysql_password` within the container.
+
+    ```bash
+    $ docker service update mysql \
+                     --secret-add source=mysql_password_2016,target=mysql_password \
+                     --secret-rm mysql_password
+    ```
+
+    Move on to the next step quickly, because WordPress clients will not be able
+    to connect to the MySQL database until you update its password.
+
+3. Update the `wordpress` service to use the new password, keeping the target
+   path at `/run/secrets/wp_db_secret` and keeping the file permissions at `0400`.
+
+   ```bash
+   $ docker service update wordpress \
+                    --secret-add source=mysql_password_2016,target=wp_db_password,mode=0400 \
+                    --secret-rm mysql_password
+   ```
+
+4.  Verify that WordPress works by going to `http://localhost:30000/` again.
+
+5.  Remove the old secret from the swarm.
+
+    ```bash
+    $ docker secret rm mysql_password
+    ```
+
+## Build support for Docker Secrets into your images
+
+If you develop a container that can be deployed as a service and accepts a
+sensitive string, such as a credential, as an environment variable, consider
+adapting your image to take advantage of Docker secrets. One way to do this is
+to ensure that each parameter you pass to the image when creating the container
+can also be read from a file.
+
+Many of the official images in the
+[Docker library](https://github.com/docker-library/), such as the
+[wordpress](https://github.com/docker-library/wordpress/)
+image used in the above examples, have been updated in this way. When you start
+a WordPress container, you provide it with the parameters it needs by setting
+them as environment variables.
+
+The WordPress image has been updated so that the environment variables which
+contain important data for WordPress, such as `WORDPRESS_DB_PASSWORD`, also have
+variants which can read their values from a file (`WORDPRESS_DB_PASSWORD_FILE`).
+This strategy ensures that backward compatibility is preserved, while allowing
+your container to read the information from a Docker-managed secret instead of
+being passed directly.
+
+>**Note**: Docker secrets do not set environment variables directly. This was
+a conscious decision, since environment variables can leak between containers
```

Comment:
I would do `since it's easy for environment variables to be unintentionally leaked`